acme/common.go

```diff
@@ -19,48 +13,59 @@ const (
 	ResourceNewOrder = Resource("new-order")
 )
 
+type Status string
```

Comment:
I think this defining of special types for status, identifier, etc. is overkill in the Boulder codebase, and I'd like to simplify it away here. Can we instead just define string-valued constants, e.g. `StatusPending = "pending"`?